pom.xml

```diff
@@ -1468,20 +1468,26 @@
 						<artifactId>build-tools</artifactId>
 						<version>${build-tools.version}</version>
 					</dependency>
+					<dependency>
+						<groupId>com.puppycrawl.tools</groupId>
+						<artifactId>checkstyle</artifactId>
+						<version>8.36.2</version>
+					</dependency>
 				</dependencies>
```

Comment:
Can we put checkstyle.xml and other configuration files to top ApromoreCore module rather than putting them inside the build-tools module and then referring to them implicitly via classpath with a dependency on the build-tools module? Checkstyle is at a project-wide scope, just don't get why it must be inside build-tools which is for extra "tools".

Reply:
Putting checkstyle.xml at the top level would work -- it's how I initially did it, until realizing how Cameron had done it better.   Doing it through the build-tools module is better because it allows components to be built using only dependencies from the Maven repo (e.g. build-tools) rather than requiring the entire ApromoreCore source tree to be present.